Platon/backend/app/search_engine/views.py

```diff
@@ -433,7 +452,124 @@ def get(self):
+    @api.expect(upcoming_events_search_parser)
+    def get(self):
+        form = UpcomingEventsSearchForm(request.args)
+        if form.validate():
+            search_query = form.search_query.data.lower()
+            # Remove the punctuation in the search query
+            search_query = SearchEngine.remove_punctuation(search_query)
+            # Tokenize the search query
+            tokens = search_query.split()
+            # Get stopwords form the API
+            stopwords = SearchEngine.get_stopwords()
+            # Remove stopwords from the list of tokens
+            tokens = list(set(tokens) - set(stopwords))
+            # Get the semantically related list for our tokens
+            tokens = SearchEngine.semantic_related_list(tokens)
+            # List of user dictionaries that will be send as a search output
+            result_list = []
+            # Score of each user record
+            result_id_score = []
+
+            # Search tokens for title, acronym, location, date or deadline match
+            for token, score in tokens:
+                try:
+                    query = '(LOWER(title) REGEXP ".*{0}.*" \
+                    OR LOWER(acronym) REGEXP ".*{0}.*" OR LOWER(location) REGEXP ".*{0}.*")'  \
+                    .format(token)
+                    sql_statement = "SELECT * FROM upcoming_events WHERE {}".format(query)
+                    result = db.engine.execute(sql_statement)
+                    for upcoming_event in result:
+                        id_list = [i[0] for i in result_id_score]
+                        if upcoming_event[0] not in id_list:
+                            if upcoming_event[-1] == 0:
+                                continue
+                            result_list.append({"id": upcoming_event[0], "title": upcoming_event[1], "acronym": upcoming_event[2],
+                                                "location": upcoming_event[3], "date": upcoming_event[4],
+                                                "deadline": upcoming_event[5], "link": upcoming_event[6]})
+                            result_id_score.append((upcoming_event[0], score))
+                        else:
+                            id_index = id_list.index(upcoming_event[0])
+                            result_id_score[id_index] = (upcoming_event[0], result_id_score[id_index][1] + score)
+                except:
+                    return make_response(jsonify({"error": "Database Connection Problem."}), 500)
 
+            # Search tokens for deadline match
+            deadline_filter_start_list = []
+            if form.deadline_filter_start.data is not None:
+                for result in result_list:
+                    event_deadline = result.get("deadline")
+                    filter_data = form.deadline_filter_start.data
+                    if datetime.strptime(event_deadline, '%b %d, %Y') >= filter_data:
+                        deadline_filter_start_list.append(result)
```

Comment:
There are some N/A's in the date parts of the Upcoming Events. So if a N/A comes to this part, the program will give an error. Please add an error handling here.